### PR TITLE
dashboard/charts

### PR DESCRIPTION
**Pull request by mezentsv**
* Added general charts component and covered with tests

closing #10
closing #11

### CODE REVIEW

File: src/app/dashboard/components/chart/chart.component.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@Component({
+  selector: 'tk-chart',
+  templateUrl: './chart.component.html'
+})
+export class ChartComponent implements OnChanges {
+  @ViewChild('chart') chart: ElementRef;
+  @Input() chartData: ChartDataType;
+  @Input() chartConfigs: ChartConfigsType;
+  @Input() chartType: 'PieChart' | 'BarChart';
```
Comment (vnazarchukGD) [inline src/app/dashboard/components/chart/chart.component.ts:23]: define ChartType enum

File: src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -1,15 +1,104 @@
...
+            product.name.toLowerCase().indexOf(search.toLowerCase()) !== -1
+        );
+      return of(searchResult || null);
+    })
+  );
+
+  readonly loading$ = new BehaviorSubject<boolean>(true);
+  private readonly destroy$ = new Subject<void>();
+
+  pieChartConfigs: ChartConfigsType = {
```
Comment (vnazarchukGD) [inline src/app/dashboard/dashboard.component.ts:49]: readonly

File: src/app/dashboard/dashboard.component.ts [hunk 3]
```diff
@@ -1,15 +1,104 @@
...
+    title: '',
+    width: 700,
+    height: 400,
+    pieSliceText: 'value'
+  };
 
-  constructor() { }
+  barChartConfigs: ChartConfigsType = {
```
Comment (vnazarchukGD) [inline src/app/dashboard/dashboard.component.ts:56]: readonly

File: src/app/dashboard/dashboard.component.ts [hunk 4]
```diff
@@ -1,15 +1,104 @@
...
+              this.barChartConfigs.title = `${name}: стоимость за единицу в зависимости от региона`;
+              break;
+            default:
+              break;
+          }
+        }
+      });
   }
 
+  ngOnInit(): void {}
```
Comment (vnazarchukGD) [inline src/app/dashboard/dashboard.component.ts:99]: redundant

File: src/app/dashboard/dashboard.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+  describe('Pie Chart', () => {
+    const targetProduct = initialState.product.products[0];
+    let pieChartComponent: ChartComponent;
+
+    beforeEach(() => {
+      store.overrideSelector(ProductSelectors.search, targetProduct.name);
+      store.refreshState();
+      fixture.detectChanges();
+
+      pieChartComponent = fixture.debugElement.queryAll(
```
Comment (vnazarchukGD) [inline src/app/dashboard/dashboard.component.spec.ts:123]: define harness for chartComponent and find pie/bar charts suing it.

File: src/index.html [hunk 6]
```diff
@@ -15,6 +15,10 @@
       href="https://fonts.googleapis.com/icon?family=Material+Icons"
       rel="stylesheet"
     />
+    <script
+      type="text/javascript"
+      src="https://www.gstatic.com/charts/loader.js"
```
Comment (vnazarchukGD) [inline src/index.html:20]: we have a library in npm for that - https://www.npmjs.com/package/angular-google-charts. 
Did you check it ? does it fit our needs ?

**Review by vnazarchukGD — CHANGES_REQUESTED** [review]

File: src/index.html (hunk 6 above)
Comment (mezentsv) [reply src/index.html:20]: Discussed it during the call

File: src/app/dashboard/dashboard.component.spec.ts (hunk 5 above)
Comment (mezentsv) [reply src/app/dashboard/dashboard.component.spec.ts:123]: I don't see a way to use a harness for this particular set of tests. The chart component is mocked and all I test is if it's rendered and if props are passed correctly. All I can do with a harness here is to check if the component is present.
Or what you suggest is to test only the presence and dump the rest of the suite as unnecessary?

Comment (vnazarchukGD) [reply src/app/dashboard/dashboard.component.spec.ts:123]: Sorry I haven't notice that you actually mocked ChartComponent. 
We are not mocking child components. in that way we can test actual component behaviour. 
But based on the way you implemented chart component, I agree, the way you locate and test it the only best possible way.

**Review by vnazarchukGD — APPROVED** [review]

**Pull request merged**